pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -653,9 +655,12 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
 
             // remove positions first from replay list first : sendMessages recycles entries
             if (readType == ReadType.Replay) {
-                entriesForThisConsumer.forEach(entry -> {
-                    redeliveryMessages.remove(entry.getLedgerId(), entry.getEntryId());
-                });
+                // we need to replay messages still present on the redeliveryMessages queue.
+                // It may happen that more than one replay is scheduled for the same message but it
+                // must be actually replayed only once.
+                entriesForThisConsumer = entriesForThisConsumer.stream()
```

Comment:
This may be a symptom that something else is broken.
We have read messages that should not have been read.

I wonder if we are replaying the messages in the wrong order.
may we can accept that because the order is here is not a matter as we are replaying messages

we should also double check the PersistentStickyKeyDispatcherMultipleConsumers, that has a different implementation of  `trySendMessagesToConsumers`